File: sync-task-scheduler/src/main/java/by/klevitov/synctaskscheduler/service/SchedulerService.java
package by.klevitov.synctaskscheduler.service;

import by.klevitov.synctaskscheduler.entity.Task;

import java.util.List;

public interface SchedulerService {
    Task scheduleTask(final Task task);

    List<Task> scheduleTask(final List<Task> tasks);

    Task rescheduleTask(final Task task);

    Task pauseTask(final Task task);

    Task resumeTask(final Task task);

    boolean deleteTask(final Task task);

    void triggerTask(final Task task);
}

File: frontend/sync-task-scheduler-ui/src/app/core/service/scheduler-service.ts
import { Injectable } from '@angular/core';
import { Task } from '../model/task';
import { SyncTaskSchedulerClient } from '../client/sync-task-scheduler-client';
import { catchError, map, Observable } from 'rxjs';
import { ErrorUtil } from '../util/error-util';

@Injectable({
  providedIn: 'root',
})
export class SchedulerService {
  constructor(private taskClient: SyncTaskSchedulerClient) {}

  triggerTask(task: Task): Observable<string> {
    return this.taskClient.triggerTask(task.id).pipe(
      map(() => 'Task was started successfully!'),
      catchError(ErrorUtil.handleError)
    );
  }

  pauseTask(task: Task): Observable<string> {
    return this.taskClient.pauseTask(task.id).pipe(
      map(() => 'Task was paused successfully!'),
      catchError(ErrorUtil.handleError)
    );
  }

  resumeTask(task: Task): Observable<string> {
    return this.taskClient.resumeTask(task.id).pipe(
      map(() => 'Task was resumed successfully!'),
      catchError(ErrorUtil.handleError)
    );
  }
}
